src/WingetCreateCLI/Commands/NewLocaleCommand.cs

```diff
@@ -0,0 +1,326 @@
+                Logger.ErrorLocalized(nameof(Resources.Error_Prefix), e.Message);
+                Logger.ErrorLocalized(nameof(Resources.OctokitNotFound_Error));
+                return false;
+            }
+            finally
+            {
+                TelemetryManager.Log.WriteEvent(commandEvent);
+            }
+        }
+
+        private static void DisplayGeneratedLocales(List<LocaleManifest> newLocales)
+        {
+            Logger.DebugLocalized(nameof(Resources.GenerateNewLocalePreview_Message));
+            foreach (var localeManifest in newLocales)
+            {
+                Logger.InfoLocalized(nameof(Resources.LocaleManifest_Message), localeManifest.PackageLocale);
+                Console.WriteLine(localeManifest.ToYaml(true));
+            }
+        }
+
+        private static void FillLocalePropertiesForUserCompletion(LocaleManifest fromManifest, LocaleManifest toManifest, List<string> properties)
+        {
+            foreach (string propertyName in properties)
+            {
+                if (propertyName == nameof(LocaleManifest.PackageLocale))
+                {
+                    continue;
+                }
+
+                var value = fromManifest.GetType().GetProperty(propertyName).GetValue(fromManifest);
+                if (value != null)
+                {
+                    toManifest.GetType().GetProperty(propertyName).SetValue(toManifest, value);
+                }
+            }
+        }
+
+        private List<LocaleManifest> ParseArgumentsAndGenerateLocales(Manifests originalManifests)
+        {
+            LocaleManifest referenceLocaleManifest;
+
+            if (!string.IsNullOrEmpty(this.ReferenceLocale))
+            {
+                try
+                {
+                    referenceLocaleManifest = (LocaleManifest)LocaleHelper.GetMatchingLocaleManifest(this.ReferenceLocale, originalManifests);
+                }
+                catch (ArgumentException)
+                {
+                    Logger.ErrorLocalized(nameof(Resources.InvalidLocale_ErrorMessage));
+                    return null;
+                }
+
+                if (referenceLocaleManifest == null)
+                {
+                    Logger.ErrorLocalized(nameof(Resources.ReferenceLocaleNotFound_Error));
+                    return null;
+                }
+            }
+            else
+            {
+                referenceLocaleManifest = originalManifests.DefaultLocaleManifest.ToLocaleManifest();
+            }
+
+            List<LocaleManifest> generatedLocales = new();
+
+            List<string> defaultPromptPropertiesForNewLocale = new()
+            {
+                nameof(LocaleManifest.PackageLocale),
+                nameof(LocaleManifest.PackageName),
+                nameof(LocaleManifest.Publisher),
+                nameof(LocaleManifest.License),
+                nameof(LocaleManifest.ShortDescription),
+            };
+
+            bool localeArgumentUsed = false;
```

Comment:
Since the new command is always interactive, is there any benefit to having this as an argument? Maybe we should just prompt the user for this to simplify things.

Reply:
I'm not sure I follow here. If you're talking about whether we should have `--locale` as an argument or not, then I would say that I tried to remain as close to the original feature request issue as possible where it was requested. As a user, I would like the flexibility to quickly enter a locale in the command and skip a prompt. I don't have strong feelings about it though, if you feel the other way, I can remove `--locale`